src/python_testing/TC_DRLK_2_9.py

```diff
@@ -0,0 +1,793 @@
+
+    async def send_set_aliro_reader_config_cmd(self, use_group_resolving_key: bool,
+                                               expected_status: Status = Status.Success):
+        try:
+
+            signingKey = bytes.fromhex("89d085fc302ca53e279bfcdecdf3c4adb2f5d9bc9ea6c49e9566d144367df3ff")
+            verificationKey = bytes.fromhex(
+                "047a4c992d753924cdf3779a3c84fec2debaa6f0b3084450878acc7ddcce7856ae57b1ebbe2561015103dd7474c2a183675378ec55f1e465ac3436bf3dd5ca54d4")
+            groupIdentifier = bytes.fromhex("89d085fc302ca53e279bfcdecdf3c4ad")
+            groupResolvingKey = bytes.fromhex("89d0859bfcdecdf3c4adfc302ca53e27")
+
+            # Checks Pics condition
+            if use_group_resolving_key is False:
+                pics_check = self.pics_guard(self.check_pics("DRLK.S.F0d") and not self.check_pics("DRLK.S.F0e") and
+                                             self.check_pics("DRLK.S.C28.Rsp"))
+            else:
+                pics_check = self.pics_guard(self.check_pics("DRLK.S.F0e") and self.check_pics("DRLK.S.C28.Rsp"))
+
+            if not use_group_resolving_key and pics_check:
+                await self.send_single_cmd(cmd=Clusters.DoorLock.Commands.SetAliroReaderConfig(
+                    signingKey=signingKey,
+                    verificationKey=verificationKey,
+                    groupIdentifier=groupIdentifier),
+                    endpoint=self.app_cluster_endpoint,
+                    timedRequestTimeoutMs=1000)
+                asserts.assert_equal(expected_status, Status.Success)
+            elif use_group_resolving_key and pics_check:
+                await self.send_single_cmd(cmd=Clusters.DoorLock.Commands.SetAliroReaderConfig(
+                    signingKey=signingKey,
+                    verificationKey=verificationKey,
+                    groupIdentifier=groupIdentifier,
+                    groupResolvingKey=groupResolvingKey),
+                    endpoint=self.app_cluster_endpoint,
+                    timedRequestTimeoutMs=1000)
+                asserts.assert_equal(expected_status, Status.Success)
+        except InteractionModelError as e:
+            logging.exception(f"Got exception when performing SetAliroReaderConfig {e}")
+            asserts.assert_equal(e.status, expected_status, f"Unexpected error returned: {e}")
+
+    @async_test_body
+    async def test_TC_DRLK_2_9(self):
+
+        self.Attrib = 0
+        numberofcredentialsupportedperuser = None
+        self.app_cluster_endpoint = 1
+        invalid_credential_type = 9
+        user_unique_id = 6459
+        user_name = "xxx"
+        credentialIndex_1 = 1
+        credentialIndex_2 = 2
+        credentialIndex_3 = 3
+        userIndex_1 = 1
+        userIndex_2 = 2
+        invalid_user_status = 5
+        invalid_user_type = 10
+
+        self.pin_code = b""
+        self.pin_code1 = b""
+        self.pin_code2 = b""
+        self.inavlid_pincode = b""
+        self.rfid_tag = b""
+
```

Comment:
Need assignments for maxpincodelength and maxrfidcodelength, eg: "self.maxpincodelength = None, self.maxrfidcodelength = None"

Reply:
Added = None assignment for the rid and pin variables